controllers/k8ssandra/reaper.go

```diff
@@ -290,3 +314,69 @@ func (r *K8ssandraClusterReconciler) addClusterToExternalReaper(
 	}
 	return result.Continue()
 }
+
+func addManagementApiSecretsToReaper(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	actualDc *cassdcapi.CassandraDatacenter,
+	logger logr.Logger,
+	authConfig *cassdcapi.ManagementApiAuthManualConfig,
+) error {
+
+	reaperName := kc.Spec.Reaper.ReaperRef.Name
+
+	reaperNamespace := kc.Spec.Reaper.ReaperRef.Namespace
+	if reaperNamespace == "" {
+		reaperNamespace = kc.Namespace
+	}
+
+	tssName := reaper.GetTruststoresSecretName(reaperName)
+	tssKey := client.ObjectKey{Namespace: reaperNamespace, Name: tssName}
+
+	tss := &corev1.Secret{}
+	if err := remoteClient.Get(ctx, tssKey, tss); err != nil {
+		logger.Error(err, "failed to get Reaper's truststore secret")
+		return err
+	}
+
+	cs := &corev1.Secret{}
+	csName := authConfig.ClientSecretName + "-ks"
```

Comment:
Issue: It looks like maybe you're re-using the same client secret currently used by cass-operator. Do you think we should maybe issue new secrets for Reaper for the sake of security?

Reply:
I agree with Micke - and not only becausae that means I don't have to change anything 😃 
Generating and managing another secret for Reaper <-> mgmt-api seems like a big headache. Taking this ad absurdum, would we then need each client of mgmt-api have its own secret - and mgmt-api accept all of them. That does not sound correct.

Reply:
The principle we have followed elsewhere is that each application gets its own set of encryption materials. So one set for Cassandra, one set for management API, one set for Reaper, etc. 

@rzvoncek suggest you check this with Chris, I'm not personally fussy either way.

And @burmanm - can you please elaborate on your perspectives here so we can understand your reasoning.

Reply:
> This is not the correct way, as mgmt-api supports a single TLS cert (combination of tls.crt/ca.crt/tls.key files, not Java keystores). Thus, if Reaper doesn't use the cass-operator one, then Reaper will not be able to connect to mgmt-api.

@burmanm I've found your reasoning in your additional comment further below. I think you've misunderstood what I'm suggesting, I am suggesting issuing a new Reaper cert using the same issuer used for the management API cert. Because they will both share the same issuer, the existing management API truststore will trust the new cert issued for Reaper.

Whether it uses a Java keystore or pem encoded stuff isn't relevant here.

Reply:
You both have valid points and it's not immediately clear how to proceed.

To move us forward, I'd like to point out that this PR is dealing with _using_ a certificate. It's not concerned what other uses that certificates have, nor where it came from. Genereating certs is mission-control, not k8ssandra, operator's job anyway.

If we decide we want dedicated certs for Reaper, we will need to track this with another ticket.

Reply:
The issue here is that you're not allowing for two different secrets to be used, I think this code assumes that the `authConfig.ClientSecretName` (which is the same secret used by management API) will always also be used by Reaper.